go/inst/instance_dao.go

```diff
@@ -1728,7 +1740,7 @@ func GetClusterOSCReplicas(clusterName string) ([](*Instance), error) {
 		return [](*Instance){}, log.Errorf("Invalid cluster name: %s", clusterName)
 	}
 	{
-		// Pick up to two busiest IMs
+		// Pick up to two busiest IMs from each DC
```

Comment:
I suspect there is also a flaw in this logic. There's an implicit assumption that all DCs all have IMs and then slaves under them.
- I know in the environments I work in that is generally the case but I've seen directly attached slaves to a master / gr_primary, and usually they are in the same DC as the master / gr_primary.  Therefore this logic may exclude these slaves if there are no IMs in the same DC as the master / gr_primary and thus ignore them, though for similar reasons they may have replication lag.  Thus these boxes should also be considered.
- e.g. we are trying to get leaf-nodes per DC and the "up to 2" most delayed leaf nodes per DC.  Current code approximates this but does not actually give us this information under some circumstances, and that may cause issues if running an OSC and wanting to throttle if delays are seen.

Reply:
> and usually they are in the same DC as the master
Isn't this case is covered by the logic at the very end:
// Get 2 1st tier leaf replicas, if possible
?

I'm not sure I understand the topology you are describing.
Should I understand it as:
For every DC get IMs (as implemented) and additionally if there are no IMs in some DC, get 2 leaves from this DC?

Reply:
It's probably best to respond directly to you on these points.